clientconn.go

```diff
@@ -1172,6 +1175,10 @@ type addrConn struct {
 	// is received, transport is closed, ac has been torn down).
 	transport transport.ClientTransport // The current transport.
 
+	// This mutex is used on the RPC path, so its usage should be minimized as
+	// much as possible.
+	// TODO: Find a lock-free way to retrieve the transport and state from the
+	// addrConn without a lock.
```

Comment:
Oops, "lock-free ... without a look" - please remove one or the other.  Thanks!

Reply:
Done. PTAL